Sources/HipMobileUI.Droid/Contracts/DroidAudioPlayer.cs

```diff
@@ -0,0 +1,135 @@
+// limitations under the License.
+
+using System;
+using System.IO;
+using System.Threading;
+using Android.Media;
+using Android.Util;
+using de.upb.hip.mobile.pcl.BusinessLayer.Models;
+using HipMobileUI.AudioPlayer;
+using Java.IO;
+using File = Java.IO.File;
+using IOException = Java.IO.IOException;
+using Stream = Android.Media.Stream;
+
+namespace de.upb.hip.mobile.droid.Contracts {
+    internal class DroidAudioPlayer : IAudioPlayer {
+
+        private readonly MediaPlayer mediaPlayer;
+        private Audio currentAudio;
+        private Timer progressUpdateTimer;
+
+        public DroidAudioPlayer ()
+        {
+            mediaPlayer = new MediaPlayer ();
+            mediaPlayer.SetAudioStreamType (Stream.Music);
+            mediaPlayer.Completion += MediaPlayerOnCompletion;
+        }
+
+        public bool IsPlaying => mediaPlayer.IsPlaying;
+
+        public double CurrentProgress => mediaPlayer.CurrentPosition;
+
+        public double MaximumProgress => mediaPlayer.Duration;
+
+        public Audio CurrentAudio {
+            get { return currentAudio; }
+            set {
+                currentAudio = value;
+                mediaPlayer.Stop ();
+                mediaPlayer.Reset ();
+                if (value != null)
+                {
+                    var path = CopyAudioToTemp (value);
+                    mediaPlayer.SetDataSource (path);
+                    mediaPlayer.Prepare ();
+                    ProgressChanged?.Invoke (0);
+                }
+            }
+        }
+
+        public event ProgressChangedDelegate ProgressChanged;
+        public event IsPlayingDelegate IsPlayingChanged;
+        public event AudioCompletedDelegate AudioCompleted;
+
+        public void Play ()
+        {
+            mediaPlayer.Start ();
+            IsPlayingChanged?.Invoke (true);
+            StartUpdateTimer ();
+        }
+
+        public void Pause ()
+        {
+            mediaPlayer.Pause ();
+            IsPlayingChanged?.Invoke(false);
+            StopUpdateTimer ();
+        }
+
+        public void Stop ()
+        {
+            mediaPlayer.Stop ();
+            IsPlayingChanged?.Invoke(false);
+            StopUpdateTimer ();
+        }
+
+        public void SeekTo (double progress)
+        {
+            mediaPlayer.SeekTo (Convert.ToInt32 (progress));
+        }
+
+        private void MediaPlayerOnCompletion (object sender, EventArgs eventArgs)
+        {
+            StopUpdateTimer ();
+            AudioCompleted?.Invoke ();
+            IsPlayingChanged?.Invoke(IsPlaying);
+        }
+
+        private void UpdateProgress (object state)
+        {
+            ProgressChanged?.Invoke (CurrentProgress);
+        }
+
+        private string CopyAudioToTemp (Audio audio)
+        {
+            var filepath = "";
+            try
+            {
+                var tempMp3 = File.CreateTempFile ("temp", ".mp3", new File (Path.GetTempPath ()));
```

Comment:
Can easily be adapted in the future. For now I don't see a benefit to generalize this as it would only mean effort without payoff. Especially, as we decided to stick with mp3 for the new content